usr/share/man/man1/nodemgr.1
.TH NODEMGR 1 2020\-04\-09
.SH NAME
nodemgr - OpenSVC agent's node management command.
.SH SYNOPSIS
.B nodemgr
om node [ \fIOPTIONS\fP ] COMMAND
.SH DESCRIPTION
In the OpenSVC infrastructure, the node is an operating system instance with an OpenSVC agent installed. The node may host OpenSVC services instances. The \fBnodemgr\fP program provides commands to configure and execute all the agent functions that interact with the node.
.SH COMMANDS

\fRCOLLECTOR MANAGEMENT\fP

  om node collector ack action

Acknowledge an action error on the node. An acknowlegment can be completed by --author (defaults to root@nodename) and --comment

.TP
.B --author=AUTHOR
The acker name to log when acknowledging action log errors
.TP
.B --comment=COMMENT
A comment to log when acknowldging action log error entries.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector alerts

Display the node alerts.

.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector asset

Display the asset information known to the collector.

.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector checks

Display the node checks.

.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector cli

Open a Command Line Interface to the collector rest API. The CLI offers autocompletion of paths and arguments, piping JSON data from files. If executed as root and with no '--user', the collector is logged in with the node credentials.

.TP
.B --user=USER
Authenticate with the collector using the specified user credentials instead of the node credentials. Required with 'om node register' when the collector is configured to refuse anonymous register.
.TP
.B --password=PASSWORD
Authenticate with the collector using the specified user credentials instead of the node credentials. Prompted if necessary but not specified.
.TP
.B --api=API
Specify a collector api url different from the one set in node.conf.
.TP
.B --insecure
Allow communications with a collector presenting an unverified SSL certificate.
.TP
.B --config=CONFIG
Specify a user-specific collector api connection configuration file. Defaults to '~/.opensvc-cli'.
.TP
.B --refresh-api
Force a reload of the collector's api digest.
.TP
.B --save
Save the collector cli settings to the file specified by --config or ~/.opensvc-cli by default.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector create tag

Create a new tag with name specified by '--tag'.

.TP
.B --tag=TAG
The tag name, as shown by 'om node collector list tags'.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector disks

Display the node disks list, complete with information issued by array parser.

.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector events

Display node events during the period specified by --begin/--end. --end defaults to now. --begin defaults to 7 days ago.

.TP
.B --begin=BEGIN
A begin date expressed as 'YYYY-MM-DD hh:mm' limiting the timerange the action applies to.
.TP
.B --end=END
A end date expressed as 'YYYY-MM-DD hh:mm' limiting the timerange the action applies to.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector list actions

List actions on the node, whatever the service, during the period specified by --begin/--end. --end defaults to now. --begin defaults to 7 days ago.

.TP
.B --begin=BEGIN
A begin date expressed as 'YYYY-MM-DD hh:mm' limiting the timerange the action applies to.
.TP
.B --end=END
A end date expressed as 'YYYY-MM-DD hh:mm' limiting the timerange the action applies to.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector list filtersets

Show the list of filtersets available on the collector. If specified, '--filterset <pattern>' limits the resultset to filtersets matching the pattern.

.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector list nodes

Show the list of nodes matching the filterset pointed by '--filterset'.

.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector list services

Show the list of services matching the filterset pointed by '--filterset'.

.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector list tags

List all available tags. Use '--like' to filter the output.

.TP
.B --like=LIKE
A data filtering expression. '%' is the multi-character wildcard. '_' is the single-character wildcard. Leading and trailing '%' are automatically set.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector log

Log a message in the collector's node log.

.TP
.B --message=MESSAGE
The message to send to the collector for logging
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector networks

Display network information known to the collector for each node ip, complete with network information from the IPAM database.

.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector search

Report the collector objects matching '--like [<type>:]<substring>', where `<type>` is the object type acronym as shown in the collector search widget.

.TP
.B --like=LIKE
A data filtering expression. '%' is the multi-character wildcard. '_' is the single-character wildcard. Leading and trailing '%' are automatically set.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector show actions

Show actions detailed log. A single action is specified by --id. a range is specified by --begin/--end dates. --end defaults to now. --begin defaults to 7 days ago.

.TP
.B --begin=BEGIN
A begin date expressed as 'YYYY-MM-DD hh:mm' limiting the timerange the action applies to.
.TP
.B --id=ID
Specify an id to act on.
.TP
.B --end=END
A end date expressed as 'YYYY-MM-DD hh:mm' limiting the timerange the action applies to.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector show tags

list all node tags

.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector tag

Set a node tag (pointed by --tag).

.TP
.B --tag=TAG
The tag name, as shown by 'om node collector list tags'.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node collector untag

Unset a node tag (pointed by --tag).

.TP
.B --tag=TAG
The tag name, as shown by 'om node collector list tags'.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

\fRCOMPLIANCE\fP

  om node compliance attach

Attach rulesets specified by '--ruleset' and modulesets specified by '--moduleset' to this node. Attached modulesets are scheduled for check or autofix.

.TP
.B --moduleset=MODULESET
Specify the modulesets to limit the action to. The special value 'all' can be used in conjonction with detach.
.TP
.B --ruleset=RULESET
Specify the rulesets to limit the action to. The special value 'all' can be used in conjonction with detach.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node compliance auto

Run compliance checks or fixes, depending on the autofix module property values.

.TP
.B --cron
If set, the action is actually executed impersonating the scheduler thread.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER
The server uri to send a request to. If not specified the local node is targeted. Supported schemes are https and raw. The default scheme is https. The default port is 1214 for the raw scheme, and 1215 for https. The uri can be a fullpath to a listener socket. In this case, the scheme is deduced from the socket. Examples: raw://1.2.3.4:1214, https://relay.opensvc.com, /var/lib/opensvc/lsnr/h2.sock.
.TP
.B --filter=JSONPATH_FILTER
A JSONPath expression to filter a JSON output.
.TP
.B --debug
Increase stream log verbosity up to the debug level.
.TP
.B --local
Set to disable cluster-wide operations.

  om node compliance check

Run compliance checks.

.TP
.B --attach
Attach the modulesets specified in a compliance run.
.TP
.B --force
Force action, ignore sanity checks.
.TP
.B --module=MODULE
Specify the modules to limit the run to. The modules must be in already attached modulesets.
.TP
.B --moduleset=MODULESET
Specify the modulesets to limit the action to. The special value 'all' can be used in conjonction with detach.
.TP
.B --ruleset-date=RULESET_DATE
Use an historical ruleset, specified by its date.
.TP
.B --node=NODE
A node selector expression. Embedded in requests for the daemon to route and multiplex the request to a list of nodes. If not specified the local node is targeted.
.TP
.B -h, --help
Show this help message and exit
.TP
.B --format=FORMAT
Specify a data formatter. Possible values are json, flat_json, csv or table. csv and table formatters are available only for commands returning tabular data.
.TP
.B --color=COLOR
Colorize output. Possible values are:

* auto: guess based on tty presence
* always|yes: always colorize
* never|no: never colorize
.TP
.B --server=SERVER

lib/nodemgr.py
from __future__ import print_function
from __future__ import absolute_import

import os
import sys

import utilities.render.color
import core.exceptions as ex
from env import Env
from nodemgr_parser import NodemgrOptParser
from core.node import Node

prog = "nodemgr"

def get_extra_argv(argv=None):
    """
    Extract extra argv from nodemgr argv.

    nodemgr can act as a wrapper for other commands (storage drivers for
    example).
    """
    if argv is None:
        argv = sys.argv[1:]
    if len(argv) < 2:
        return argv, []

    if "array" in argv:
        pos = argv.index('array')
    elif "cli" in argv:
        pos = argv.index('cli')
        if pos > 0 and argv[pos-1] != "collector":
            return argv, []
    else:
        return argv, []

    if "--" in argv:
        pos = argv.index("--")
    if len(argv) > pos + 1:
        extra_argv = argv[pos+1:]
    else:
        extra_argv = []
    argv = argv[:pos+1]
    return argv, extra_argv

def do_symcli_db_file(options):
    try:
        symcli_db_file = options.symcli_db_file
    except AttributeError:
        return
    if symcli_db_file is None:
        return
    if not os.path.exists(symcli_db_file):
        print("File does not exist: %s" % symcli_db_file)
        return
    os.environ['SYMCLI_DB_FILE'] = symcli_db_file
    os.environ['SYMCLI_OFFLINE'] = '1'

def _main(node, argv=None):
    argv, extra_argv = get_extra_argv(argv)
    optparser = NodemgrOptParser(argv)
    options, action = optparser.parse_args(argv)
    options.extra_argv = extra_argv

    utilities.render.color.use_color = options.color
    node.options.update(options.__dict__)
    do_symcli_db_file(options)

    if action.startswith("collector_cli"):
        action = "collector_cli"

    node.check_privs(action)

    err = 0
    try:
        err = node.action(action)
    except KeyboardInterrupt:
        sys.stderr.write("Keybord Interrupt\n")
        err = 1
    except ex.Error:
        import traceback
        exc_type, exc_value, exc_traceback = sys.exc_info()
        es = str(exc_value)
        if len(es) > 0:
            sys.stderr.write(str(exc_value)+'\n')
        err = 1
    except:
        raise
        err = 1
    return err

def main(argv=None):
    node = Node()

    try:
        return _main(node, argv=argv)
    except ex.Error as exc:
        print(exc, file=sys.stderr)
        return 1
    except ex.Version as exc:
        print(exc)
        return 0
    finally:
        node.close()

if __name__ == "__main__":
    ret = main()
    sys.exit(ret)

